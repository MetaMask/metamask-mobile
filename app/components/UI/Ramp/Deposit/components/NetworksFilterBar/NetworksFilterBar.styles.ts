import { StyleSheet } from 'react-native';

const styleSheet = () =>
  StyleSheet.create({
    networksContainer: {
      paddingHorizontal: 16,
      flexDirection: 'row',
      gap: 8,
    },
    selectedNetworkIcon: {
      marginRight: 8,
    },
    overlappedNetworkIcon: {
      marginLeft: -6,
    },
  });

export default styleSheet;
