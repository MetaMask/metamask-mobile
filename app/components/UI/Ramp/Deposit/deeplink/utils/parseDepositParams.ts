import { DepositNavigationParams } from '../../types';

export default function parseDepositParams(
  depositParams: Record<string, string | undefined>,
): DepositNavigationParams | undefined {
  // create a rampIntent object if the pathParams contain the necessary fields
  const params: DepositNavigationParams = {
    assetId: depositParams.assetId,
    amount: depositParams.amount,
  };

  if (!params.assetId && !params.amount) {
    return undefined;
  }

  Object.keys(params).forEach(
    (key) =>
      params[key as keyof DepositNavigationParams] === undefined &&
      delete params[key as keyof DepositNavigationParams],
  );

  return params;
}
