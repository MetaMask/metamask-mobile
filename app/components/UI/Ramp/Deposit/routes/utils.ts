import { DepositNavigationParams } from '../types';
import Routes from '../../../../../constants/navigation/Routes';

export function createDepositNavigationDetails(
  params?: DepositNavigationParams,
) {
  const route = Routes.DEPOSIT.ID;
  if (!params) {
    return [route] as const;
  }
  return [route, { screen: route, params }] as const;
}
