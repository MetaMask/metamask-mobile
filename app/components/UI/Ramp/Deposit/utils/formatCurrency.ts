import I18n from '../../../../../../locales/i18n';
import { getIntlNumberFormatter } from '../../../../../util/intl';

/**
 * Formats currency amounts using the device's locale
 * @param amount - The amount to format (number or string)
 * @param currency - The currency code (e.g., 'USD', 'EUR')
 * @param options - Additional formatting options
 * @returns Formatted currency string
 */

export function formatCurrency(
  amount: number | string,
  currency: string,
  options?: Intl.NumberFormatOptions,
): string {
  try {
    const numAmount = typeof amount === 'string' ? parseFloat(amount) : amount;
    const defaultOptions: Intl.NumberFormatOptions = {
      style: 'currency',
      currency: currency || 'USD',
      currencyDisplay: 'symbol',
    };

    return getIntlNumberFormatter(I18n.locale, {
      ...defaultOptions,
      ...options,
    }).format(numAmount);
  } catch (error) {
    console.error('Error formatting currency:', error);
    return amount.toString();
  }
}
