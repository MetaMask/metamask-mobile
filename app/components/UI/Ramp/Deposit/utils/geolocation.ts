import { TRANSAK_API_URL, TRANSAK_GEOLOCATION_ENDPOINT } from '../constants';

export interface GeolocationResponse {
  ipCountryCode: string;
}

export async function getGeolocation(): Promise<GeolocationResponse> {
  if (!TRANSAK_API_URL) {
    throw new Error('Transak API URL is not set');
  }

  const myURL = new URL(TRANSAK_GEOLOCATION_ENDPOINT, TRANSAK_API_URL);

  const response = await fetch(myURL.toString(), {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  });

  if (!response.ok) {
    throw new Error(
      `Transak geolocation API request failed with status: ${response.status}`,
    );
  }

  const data = await response.json();

  if (!data.ipCountryCode) {
    throw new Error('Invalid geolocation response: missing ipCountryCode');
  }

  return data;
}
