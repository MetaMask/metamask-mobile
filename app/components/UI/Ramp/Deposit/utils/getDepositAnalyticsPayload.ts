import { FiatOrder } from '../../../../../reducers/fiatOrders';

function getDepositAnalyticsPayload(_fiatOrder: FiatOrder) {
  // TODO: Implement the logic for generating analytics payload for deposit orders.
  return [null, null];
}
export default getDepositAnalyticsPayload;
