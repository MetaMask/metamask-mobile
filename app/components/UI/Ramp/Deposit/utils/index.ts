'use strict';

const emailRegex =
  /^[-!#$%&'*+/0-9=?A-Z^_a-z`{|}~](\.?[-!#$%&'*+/0-9=?A-Z^_a-z`{|}~])*@[a-zA-Z0-9](-*\.?[a-zA-Z0-9])*\.[a-zA-Z](-?[a-zA-Z0-9])+$/;

export const validateEmail = function (email: string) {
  if (!email || email.split('@').length !== 2) return false;
  return emailRegex.test(email);
};

export const formatUSPhoneNumber = (text: string) => {
  const cleaned = text.replace(/\D/g, '');
  if (cleaned.length === 0) return '';
  if (cleaned.length <= 3) {
    return `(${cleaned}`;
  } else if (cleaned.length <= 6) {
    return `(${cleaned.slice(0, 3)}) ${cleaned.slice(3)}`;
  }
  return `(${cleaned.slice(0, 3)}) ${cleaned.slice(3, 6)}-${cleaned.slice(
    6,
    10,
  )}`;
};

import {
  DepositCryptoCurrency,
  DepositFiatCurrency,
  DepositPaymentMethod,
} from '../constants';

const TRANSAK_CRYPTO_IDS: Record<string, string> = {
  'eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48': 'USDC',
  'eip155:1/erc20:0xdAC17F958D2ee523a2206206994597C13D831ec7': 'USDT',
};

const TRANSAK_FIAT_IDS: Record<string, string> = {
  USD: 'USD',
  EUR: 'EUR',
};

const TRANSAK_CHAIN_IDS: Record<string, string> = {
  'eip155:1': 'ethereum',
};

const TRANSAK_PAYMENT_METHOD_IDS: Record<string, string> = {
  credit_debit_card: 'credit_debit_card',
};

/**
 * Transforms our internal crypto currency ID to Transak crypto currency ID
 * @param cryptoCurrency - The crypto currency object
 * @returns The Transak crypto currency ID
 */
export function getTransakCryptoCurrencyId(
  cryptoCurrency: DepositCryptoCurrency,
): string {
  const transakId = TRANSAK_CRYPTO_IDS[cryptoCurrency.assetId];
  if (!transakId) {
    throw new Error(`Unsupported crypto currency: ${cryptoCurrency.assetId}`);
  }
  return transakId;
}

/**
 * Transforms our internal fiat currency ID to Transak fiat currency ID
 * @param fiatCurrency - The fiat currency object
 * @returns The Transak fiat currency ID
 */
export function getTransakFiatCurrencyId(
  fiatCurrency: DepositFiatCurrency,
): string {
  const transakId = TRANSAK_FIAT_IDS[fiatCurrency.id];
  if (!transakId) {
    throw new Error(`Unsupported fiat currency: ${fiatCurrency.id}`);
  }
  return transakId;
}

/**
 * Transforms our internal chain ID to Transak chain ID
 * @param chainId - The chain ID
 * @returns The Transak chain ID
 */
export function getTransakChainId(chainId: string): string {
  const transakId = TRANSAK_CHAIN_IDS[chainId];
  if (!transakId) {
    throw new Error(`Unsupported chain: ${chainId}`);
  }
  return transakId;
}

/**
 * Transforms our internal payment method ID to Transak payment method ID
 * @param paymentMethod - The payment method object
 * @returns The Transak payment method ID
 */
export function getTransakPaymentMethodId(
  paymentMethod: DepositPaymentMethod,
): string {
  const transakId = TRANSAK_PAYMENT_METHOD_IDS[paymentMethod.id];
  if (!transakId) {
    throw new Error(`Unsupported payment method: ${paymentMethod.id}`);
  }
  return transakId;
}
