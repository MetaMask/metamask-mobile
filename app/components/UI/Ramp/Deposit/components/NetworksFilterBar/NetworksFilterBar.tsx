import React, { useMemo } from 'react';
import { useSelector } from 'react-redux';
import { CaipChainId } from '@metamask/utils';
import { ScrollView } from 'react-native-gesture-handler';

import { AvatarSize } from '../../../../../../component-library/components/Avatars/Avatar';
import AvatarNetwork from '../../../../../../component-library/components/Avatars/Avatar/variants/AvatarNetwork';
import { Box } from '../../../../Box/Box';
import { AlignItems, FlexDirection } from '../../../../Box/box.types';
import Button, {
  ButtonSize,
  ButtonVariants,
} from '../../../../../../component-library/components/Buttons/Button';
import Text, {
  TextColor,
  TextVariant,
} from '../../../../../../component-library/components/Texts/Text';
import Icon, {
  IconName,
  IconSize,
} from '../../../../../../component-library/components/Icons/Icon';

import styleSheet from './NetworksFilterBar.styles';

import { selectNetworkConfigurationsByCaipChainId } from '../../../../../../selectors/networkController';
import { useStyles } from '../../../../../hooks/useStyles';
import { DEPOSIT_NETWORKS_BY_CHAIN_ID } from '../../constants/networks';
import { excludeFromArray } from '../../utils';
import { getNetworkImageSource } from '../../../../../../util/networks';
import { useTheme } from '../../../../../../util/theme';
import { strings } from '../../../../../../../locales/i18n';

interface NetworksFilterBarProps {
  networks: CaipChainId[];
  networkFilter: CaipChainId[] | null;
  setNetworkFilter: React.Dispatch<React.SetStateAction<CaipChainId[] | null>>;
  setIsEditingNetworkFilter: (isEditing: boolean) => void;
}

function NetworksFilterBar({
  networks,
  networkFilter,
  setNetworkFilter,
  setIsEditingNetworkFilter,
}: Readonly<NetworksFilterBarProps>) {
  const { styles } = useStyles(styleSheet, {});
  const { colors } = useTheme();

  const allNetworkConfigurations = useSelector(
    selectNetworkConfigurationsByCaipChainId,
  );

  const allNetworksIcons = useMemo(() => {
    const headSize = 3;
    const reversedHead = networks.slice(0, headSize).reverse();
    return (
      <Box flexDirection={FlexDirection.RowReverse} gap={0}>
        {reversedHead.map((chainId) => (
          <AvatarNetwork
            key={chainId}
            imageSource={getNetworkImageSource({ chainId })}
            name={allNetworkConfigurations[chainId]?.name}
            size={AvatarSize.Xs}
            style={styles.overlappedNetworkIcon}
            includesBorder
          />
        ))}
      </Box>
    );
  }, [allNetworkConfigurations, styles.overlappedNetworkIcon, networks]);

  return (
    <ScrollView
      horizontal
      showsHorizontalScrollIndicator={false}
      contentContainerStyle={styles.networksContainer}
    >
      {networkFilter && networkFilter.length !== networks.length ? (
        <>
          <Button
            variant={ButtonVariants.Secondary}
            size={ButtonSize.Sm}
            label={
              <Box
                flexDirection={FlexDirection.Row}
                alignItems={AlignItems.center}
                gap={8}
              >
                <Text variant={TextVariant.BodyMD}>See all</Text>
                <Icon
                  name={IconName.ArrowDown}
                  size={IconSize.Xs}
                  color={colors.icon.default}
                />
              </Box>
            }
            onPress={() => setIsEditingNetworkFilter(true)}
          />
          {networks.map((chainId) => {
            const isSelected = networkFilter.includes(chainId);
            const networkName =
              DEPOSIT_NETWORKS_BY_CHAIN_ID[chainId]?.name ??
              allNetworkConfigurations[chainId]?.name;
            return (
              <Button
                key={chainId}
                variant={
                  isSelected ? ButtonVariants.Primary : ButtonVariants.Secondary
                }
                size={ButtonSize.Sm}
                label={
                  <>
                    {isSelected && (
                      <AvatarNetwork
                        key={chainId}
                        imageSource={getNetworkImageSource({ chainId })}
                        name={networkName}
                        size={AvatarSize.Xs}
                        style={styles.selectedNetworkIcon}
                      />
                    )}
                    <Text
                      color={isSelected ? TextColor.Inverse : TextColor.Default}
                    >
                      {networkName}
                    </Text>
                  </>
                }
                onPress={() => {
                  if (isSelected && networkFilter.length > 1) {
                    setNetworkFilter((prev) =>
                      excludeFromArray(prev || [], chainId),
                    );
                  } else {
                    setNetworkFilter([chainId]);
                  }
                }}
              />
            );
          })}
        </>
      ) : (
        <Button
          variant={ButtonVariants.Secondary}
          size={ButtonSize.Sm}
          label={
            <Box
              flexDirection={FlexDirection.Row}
              alignItems={AlignItems.center}
              gap={8}
            >
              {allNetworksIcons}
              <Text variant={TextVariant.BodyMD}>
                {strings('deposit.networks_filter_bar.all_networks')}
              </Text>
              <Icon
                name={IconName.ArrowDown}
                size={IconSize.Xs}
                color={colors.icon.default}
              />
            </Box>
          }
          onPress={() => {
            setNetworkFilter(networks);
            setIsEditingNetworkFilter(true);
          }}
        />
      )}
    </ScrollView>
  );
}

export default NetworksFilterBar;
