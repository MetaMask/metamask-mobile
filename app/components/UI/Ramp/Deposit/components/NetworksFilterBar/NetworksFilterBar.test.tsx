import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import NetworksFilterBar from './NetworksFilterBar';
import { CaipChainId } from '@metamask/utils';
import renderWithProvider from '../../../../../../util/test/renderWithProvider';
import initialRootState from '../../../../../../util/test/initial-root-state';

function render(component: React.ReactElement) {
  return renderWithProvider(component, {
    state: initialRootState,
  });
}

describe('NetworksFilterBar', () => {
  const networks: CaipChainId[] = ['eip155:1', 'eip155:59144', 'eip155:56'];
  const mockSetNetworkFilter = jest.fn();
  const mockSetIsEditingNetworkFilter = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when networkFilter is null', () => {
    const { toJSON, getByText } = render(
      <NetworksFilterBar
        networks={networks}
        networkFilter={null}
        setNetworkFilter={mockSetNetworkFilter}
        setIsEditingNetworkFilter={mockSetIsEditingNetworkFilter}
      />,
    );

    expect(toJSON()).toMatchSnapshot();

    const allNetworksButton = getByText('All networks');
    fireEvent.press(allNetworksButton);

    expect(mockSetNetworkFilter).toHaveBeenCalledWith(networks);
    expect(mockSetIsEditingNetworkFilter).toHaveBeenCalledWith(true);
  });

  it('renders correctly when networkFilter is the same as uniqueNetworks', () => {
    const { toJSON, getByText } = render(
      <NetworksFilterBar
        networks={networks}
        networkFilter={networks}
        setNetworkFilter={mockSetNetworkFilter}
        setIsEditingNetworkFilter={mockSetIsEditingNetworkFilter}
      />,
    );

    expect(toJSON()).toMatchSnapshot();

    const allNetworksButton = getByText('All networks');
    fireEvent.press(allNetworksButton);

    expect(mockSetNetworkFilter).toHaveBeenCalledWith(networks);
    expect(mockSetIsEditingNetworkFilter).toHaveBeenCalledWith(true);
  });

  it('renders correctly when networkFilter is a subset of uniqueNetworks and removes selection', () => {
    const subsetNetworkFilter: CaipChainId[] = ['eip155:1', 'eip155:59144'];
    const { toJSON, getByText } = render(
      <NetworksFilterBar
        networks={networks}
        networkFilter={subsetNetworkFilter}
        setNetworkFilter={mockSetNetworkFilter}
        setIsEditingNetworkFilter={mockSetIsEditingNetworkFilter}
      />,
    );

    expect(toJSON()).toMatchSnapshot();

    const ethereumButton = getByText('Ethereum');
    fireEvent.press(ethereumButton);

    expect(mockSetNetworkFilter).toHaveBeenCalled();
    const setterFunction = mockSetNetworkFilter.mock.calls[0][0];
    expect(setterFunction(subsetNetworkFilter)).toEqual(['eip155:59144']);
  });

  it('replaces networkFilter when only one network is selected', () => {
    const singleNetworkFilter: CaipChainId[] = ['eip155:1'];
    const { getByText } = render(
      <NetworksFilterBar
        networks={networks}
        networkFilter={singleNetworkFilter}
        setNetworkFilter={mockSetNetworkFilter}
        setIsEditingNetworkFilter={mockSetIsEditingNetworkFilter}
      />,
    );

    const lineaButton = getByText('Linea');
    fireEvent.press(lineaButton);

    expect(mockSetNetworkFilter).toHaveBeenCalledWith(['eip155:59144']);
  });
});
