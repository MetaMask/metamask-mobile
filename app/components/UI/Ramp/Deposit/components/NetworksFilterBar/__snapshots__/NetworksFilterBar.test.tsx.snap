// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworksFilterBar renders correctly when networkFilter is a subset of uniqueNetworks and removes selection 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    {
      "flexDirection": "row",
      "gap": 8,
      "paddingHorizontal": 16,
    }
  }
  handlerTag={-1}
  handlerType="NativeViewGestureHandler"
  horizontal={true}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  showsHorizontalScrollIndicator={false}
>
  <View>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        alignItems="center"
        flexDirection="row"
        gap={8}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Regular",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          See all
        </Text>
        <SvgMock
          color="#121314"
          fill="currentColor"
          height={12}
          name="ArrowDown"
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
          width={12}
        />
      </View>
    </TouchableOpacity>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#121314",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "height": 16,
            "justifyContent": "center",
            "marginRight": 8,
            "overflow": "hidden",
            "width": 16,
          }
        }
      >
        <Image
          onError={[Function]}
          resizeMode="contain"
          source={1}
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
          testID="network-avatar-image"
        />
      </View>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#ffffff",
            "fontFamily": "Geist Regular",
            "fontSize": 16,
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        Ethereum
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#121314",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 4,
            "height": 16,
            "justifyContent": "center",
            "marginRight": 8,
            "overflow": "hidden",
            "width": 16,
          }
        }
      >
        <Image
          onError={[Function]}
          resizeMode="contain"
          source={1}
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
          testID="network-avatar-image"
        />
      </View>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#ffffff",
            "fontFamily": "Geist Regular",
            "fontSize": 16,
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        Linea
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#121314",
            "fontFamily": "Geist Regular",
            "fontSize": 16,
            "letterSpacing": 0,
            "lineHeight": 24,
          }
        }
      >
        BNB Smart Chain
      </Text>
    </TouchableOpacity>
  </View>
</RCTScrollView>
`;

exports[`NetworksFilterBar renders correctly when networkFilter is null 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    {
      "flexDirection": "row",
      "gap": 8,
      "paddingHorizontal": 16,
    }
  }
  handlerTag={-1}
  handlerType="NativeViewGestureHandler"
  horizontal={true}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  showsHorizontalScrollIndicator={false}
>
  <View>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        alignItems="center"
        flexDirection="row"
        gap={8}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
            },
            undefined,
          ]
        }
      >
        <View
          flexDirection="row-reverse"
          gap={0}
          style={
            [
              {
                "flexDirection": "row-reverse",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
        </View>
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Regular",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          All networks
        </Text>
        <SvgMock
          color="#121314"
          fill="currentColor"
          height={12}
          name="ArrowDown"
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
          width={12}
        />
      </View>
    </TouchableOpacity>
  </View>
</RCTScrollView>
`;

exports[`NetworksFilterBar renders correctly when networkFilter is the same as uniqueNetworks 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    {
      "flexDirection": "row",
      "gap": 8,
      "paddingHorizontal": 16,
    }
  }
  handlerTag={-1}
  handlerType="NativeViewGestureHandler"
  horizontal={true}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  showsHorizontalScrollIndicator={false}
>
  <View>
    <TouchableOpacity
      accessibilityRole="button"
      accessible={true}
      activeOpacity={1}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        alignItems="center"
        flexDirection="row"
        gap={8}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8,
            },
            undefined,
          ]
        }
      >
        <View
          flexDirection="row-reverse"
          gap={0}
          style={
            [
              {
                "flexDirection": "row-reverse",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#ffffff",
                "borderRadius": 4,
                "borderWidth": 1.5,
                "height": 16,
                "justifyContent": "center",
                "marginLeft": -6,
                "overflow": "hidden",
                "width": 16,
              }
            }
          >
            <Image
              onError={[Function]}
              resizeMode="contain"
              source={1}
              style={
                {
                  "height": 16,
                  "width": 16,
                }
              }
              testID="network-avatar-image"
            />
          </View>
        </View>
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#121314",
              "fontFamily": "Geist Regular",
              "fontSize": 16,
              "letterSpacing": 0,
              "lineHeight": 24,
            }
          }
        >
          All networks
        </Text>
        <SvgMock
          color="#121314"
          fill="currentColor"
          height={12}
          name="ArrowDown"
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
          width={12}
        />
      </View>
    </TouchableOpacity>
  </View>
</RCTScrollView>
`;
